packages/server/tests/UserVoteForTrack.test.ts

```diff
@@ -85,68 +91,95 @@ test.group(`User service socket handler tests`, (group) => {
-            receivedEvents: [] as string[],
-        };
+        const userCSocket = await createAuthenticatedUserAndGetSocket({
+            userID: userCID,
+            mtvRoomIDToAssociate: roomID,
+        });
+        const userCToken = getSocketApiAuthToken(userCSocket);
+        const userCSocketB = await createSocketConnection({
+            userID: userCID,
+            token: userCToken,
+        });
 
-        creatorSocket.socket.once('MTV_VOTE_OR_SUGGEST_TRACK_CALLBACK', () =>
-            creatorSocket.receivedEvents.push(
+        const creatorSocketVoteOrSuggestCallbackSpy =
+            createSpyOnClientSocketEvent(
+                creatorSocket,
                 'MTV_VOTE_OR_SUGGEST_TRACK_CALLBACK',
-            ),
+            );
+        const creatorSocketBVoteOrSuggestCallbackSpy =
+            createSpyOnClientSocketEvent(
+                creatorSocketB,
+                'MTV_VOTE_OR_SUGGEST_TRACK_CALLBACK',
+            );
+
+        const creatorSocketListUpdate = createSpyOnClientSocketEvent(
+            creatorSocket,
+            'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
         );
 
-        creatorSocket.socketB.once('MTV_VOTE_OR_SUGGEST_TRACK_CALLBACK', () =>
-            creatorSocket.receivedEvents.push(
-                'MTV_VOTE_OR_SUGGEST_TRACK_CALLBACK',
-            ),
+        const creatorSocketBListUpdate = createSpyOnClientSocketEvent(
+            creatorSocketB,
+            'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
         );
 
-        const members = [creatorSocket, userBSocket, userCSocket];
-        members.forEach((socket) => {
-            socket.socket.once('MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE', () =>
-                socket.receivedEvents.push(
-                    'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
-                ),
-            );
+        const userBSocketListUpdate = createSpyOnClientSocketEvent(
+            userBSocket,
+            'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
+        );
 
-            socket.socketB.once('MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE', () =>
-                socket.receivedEvents.push(
-                    'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
-                ),
-            );
-        });
+        const userBSocketBListUpdate = createSpyOnClientSocketEvent(
+            userBSocketB,
+            'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
+        );
 
-        creatorSocket.socket.emit('MTV_VOTE_FOR_TRACK', {
+        const userCSocketListUpdate = createSpyOnClientSocketEvent(
+            userCSocket,
+            'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
+        );
+
+        const userCSocketBListUpdate = createSpyOnClientSocketEvent(
+            userCSocketB,
+            'MTV_VOTE_OR_SUGGEST_TRACKS_LIST_UPDATE',
+        );
+
+        creatorSocket.emit('MTV_VOTE_FOR_TRACK', {
             trackID: datatype.uuid(),
         });
-        await sleep();
-        await sleep();
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Ouch! You're right to use `waitFor` instead!